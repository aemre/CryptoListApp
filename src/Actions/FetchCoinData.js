import { apiBaseURL } from "./../Utils/Constants";
import {
  FETCHING_COIN_DATA,
  FETCHING_COIN_DATA_SUCCESS,
  FETCHING_COIN_DATA_FAIL
} from "./../Utils/ActionTypes";
const options = {
  method: "GET",
  uri: "https://pro-api.coinmarketcap.com/v1/cryptocurrency/listings/latest",
  qs: {
    start: "1",
    limit: "1",
    convert: "USD,BTC"
  },
  headers: {
    "X-CMC_PRO_API_KEY": "2ba1058f-b5dc-4ddb-8e5c-0e1de7c820ab"
  },
  json: true,
  gzip: true
};

export const FetchCoinData = () => dispatch => {
  dispatch({ type: FETCHING_COIN_DATA });

  return fetch(apiBaseURL, options)
    .then(res => res.json())

    .then(x => {
      return dispatch({ type: FETCHING_COIN_DATA_SUCCESS, payload: x.data });
    })
    .catch(err => {
      return dispatch({ type: FETCHING_COIN_DATA_FAIL, payload: err });
    });
};

export function testing() {
  console.warn("testing");
}
