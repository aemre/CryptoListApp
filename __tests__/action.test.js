import FetchCoinData, { aFetchCoinData } from "../src/Actions/FetchCoinData";
import {
  FETCHING_COIN_DATA,
  FETCHING_COIN_DATA_SUCCESS,
  FETCHING_COIN_DATA_FAIL,
  API_CALL_RESTORE
} from "../src/Utils/ActionTypes";

import configureMockStore from "redux-mock-store";
import thunk from "redux-thunk";

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);
const store = mockStore({});

describe("select_actions", () => {
  beforeEach(() => {
    // Runs before each test in the suite
    store.clearActions();
  });
  describe("selectAvatar", () => {
    test("Dispatches the correct action and payload", () => {
      store.dispatch(functionToTest());
      expect(store.getActions()).toMatchSnapshot();
    });
  });
});

export const functionToTest = () => async dispatch => {
  dispatch(FetchCoinData());
};
