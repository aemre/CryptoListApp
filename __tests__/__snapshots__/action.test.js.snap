// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`select_actions selectAvatar Dispatches the correct action and payload 1`] = `Array []`;
